src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -767,9 +678,8 @@ trait ChoosesMode {
 
     val remainingTripData: Option[ParkingMNL.RemainingTripData] =
       stateData match {
-        case _: BasePersonData =>
-          calculateRemainingTripData(stateData.asInstanceOf[BasePersonData])
-        case _ => None
+        case basePersonData: BasePersonData => calculateRemainingTripData(basePersonData)
```

Comment:
Without running this I'm unsure if erasure plays into this and will result in a warning? But we should have warnings as errors still...so it's probably fine.